### PR TITLE
Java api use apply boolean mask

### PR DESCRIPTION
**Pull request by abellina**
This is to mirror the change for `apply_bolean_mask` to use tables that was merged here: https://github.com/rapidsai/cudf/pull/1744 for the java api filter operations.

### CODE REVIEW

File: java/src/main/java/ai/rapids/cudf/Table.java [hunk 1]
```diff
@@ -213,6 +213,9 @@ public String toString() {
 
   private static native long[] concatenate(long[] cudfTablePointers) throws CudfException;
 
+  private static native long[] gdfFilter(long input, long mask);
+
+
```
Comment (kuhushukla) [inline java/src/main/java/ai/rapids/cudf/Table.java]: intentional 2 extra newlines?

File: java/src/test/java/ai/rapids/cudf/TableTest.java [hunk 2]
```diff
@@ -1163,4 +1163,121 @@ void testMultiAgg() {
       }
     }
   }
+
+  //
+  // FILTER TESTS
+  //
```
Comment (kuhushukla) [inline java/src/test/java/ai/rapids/cudf/TableTest.java]: Can be removed along with the follwoing new line IMO

File: java/src/main/java/ai/rapids/cudf/Table.java (hunk 1 above)
Comment (abellina) [reply java/src/main/java/ai/rapids/cudf/Table.java]: @kuhushukla, made it a single line.

File: java/src/test/java/ai/rapids/cudf/TableTest.java (hunk 2 above)
Comment (abellina) [reply java/src/test/java/ai/rapids/cudf/TableTest.java]: done.

**Comment by tgravescs** [comment]
+1 looks good

**Pull request merged**